konig-schemagen/src/main/java/io/konig/schemagen/gcp/GoogleAnalyticsUdfGenerator.java

```diff
@@ -0,0 +1,186 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.List;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableSchema;
+
+import io.konig.core.KonigException;
+import io.konig.core.io.ShapeFileFactory;
+import io.konig.core.vocab.Konig;
+import io.konig.datasource.DataSource;
+import io.konig.gcp.datasource.GoogleBigQueryTable;
+import io.konig.shacl.Shape;
+import io.konig.shacl.ShapeManager;
+import io.konig.shacl.ShapeVisitor;
+
+public class GoogleAnalyticsUdfGenerator implements ShapeVisitor {
+	
+
+	private ShapeFileFactory fileFactory;
+	private ShapeManager shapeManager;
+	
+	public GoogleAnalyticsUdfGenerator(ShapeFileFactory fileFactory,ShapeManager shapeManager) {
+		this.fileFactory = fileFactory;
+		this.shapeManager = shapeManager;
+	}
+	public GoogleAnalyticsUdfGenerator(ShapeFileFactory fileFactory) {
+		this.fileFactory = fileFactory;
+	}
+
+	@Override
+	public void visit(Shape shape) {
+		// Get a description of the target BigQuery table (i.e. the destination table)
+		GoogleBigQueryTable targetTable = getTargetTable(shape);		
+		if (targetTable != null) {	
+			BigQueryTableGenerator tableGenerator = new BigQueryTableGenerator(shapeManager);
+			TableSchema schema = tableGenerator.toTableSchema(shape);
+			String function = createQuery(schema)
+					.replace("@destinationTable", targetTable.getTableIdentifier())
+					.replace("@gaDataset", "131116259");
+			File file = fileFactory.createFile(shape);
+			writeResources(file, function);			
+		}
+		
+	}
+	
+	private void createTempFunction(StringBuilder builder, List<TableFieldSchema> fieldsSchema) {
+		builder.append("CREATE TEMP FUNCTION parseJson(json_str STRING) \n");
+		builder.append("RETURNS STRUCT< \n");
+		writeOutputSchema(builder, fieldsSchema);
+		builder.append("> \n");
+		builder.append("LANGUAGE js AS \"\"\" \n");
+		builder.append("var row = JSON.parse(json_str); \n");
+		builder.append("row['eventTime'] = new Date(row.eventTime); \n");
+		builder.append("return row; \n");
+		builder.append(" \"\"\"; \n");
+	}
+	
+	private void doParsing(StringBuilder builder) {		
+		builder.append("Select parseJson(h.eventInfo.eventLabel) as output \n");
+		builder.append("from `@gaDataset.ga_sessions_*`,UNNEST(hits) as h \n");
+		builder.append("where (_TABLE_SUFFIX > @fromDate "
+				+ "AND _TABLE_SUFFIX < @toDate) "
+				+ "AND h.eventInfo.eventCategory = @eventCategory \n");
```

Comment:
I now understand that the '@' syntax is supported by the Google API.  Makes sense to use the '@' symbol to replace query parameters since it is supported by the Google API.